File: docs-examples/example-config-java/src/main/java/io/micronaut/configuration/openapi/docs/config/UserOperationsController.java
package io.micronaut.configuration.openapi.docs.config;

// tag::imports[]
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Post;
// end::imports[]

// tag::clazz[]
/**
 * User main operations.
 */
@Controller
public class UserOperationsController {

    @Get("/read/{id}")
    public String read(String id) {
        return "OK!";
    }

    @Post("/save/{id}")
    public String save2(String id, Object body) {
        return "OK!";
    }

    @Post("/save")
    public String save(Object body) {
        return "OK!";
    }
}
// end::clazz[]

File: docs-examples/example-config-kotlin/src/main/kotlin/io/micronaut/configuration/openapi/docs/config/UserOperationsController.kt
package io.micronaut.configuration.openapi.docs.config

// tag::imports[]
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Post
// end::imports[]

// tag::clazz[]
/**
 * User main operations.
 */
@Controller
class UserOperationsController {

    @Get("/read/{id}")
    fun read(id: String): String = "OK!"

    @Post("/save/{id}")
    fun save2(id: String, body: Any): String = "OK!"

    @Post("/save")
    fun save(body: Any): String = "OK!"
}
// end::clazz[]
